villain/parse.rkt

```diff
@@ -58,6 +54,17 @@
     [(cons 'quote (list (? symbol? x))) (Symbol x)]
     [(list 'match e0 cs ...)
      (Match (parse-e e0) (map parse-c cs))]
+    [(list-rest 'letrec bs e es)
+     (let ((x+es (map parse-binding bs)))
+       (Letrec (map first x+es) (map second x+es) (parse-seq e es)))]
+    [(or (list-rest 'lambda (list xs ...) e es)
```

Comment:
The `or` can be pushed in to `(or 'lambda 'λ)`.

Reply:
Will make the change. Thanks!